src/FNA3D_Memory.h

```diff
@@ -0,0 +1,253 @@
+/* FNA3D - 3D Graphics Library for FNA
+ *
+ * Copyright (c) 2020-2023 Ethan Lee
+ *
+ * This software is provided 'as-is', without any express or implied warranty.
+ * In no event will the authors be held liable for any damages arising from
+ * the use of this software.
+ *
+ * Permission is granted to anyone to use this software for any purpose,
+ * including commercial applications, and to alter it and redistribute it
+ * freely, subject to the following restrictions:
+ *
+ * 1. The origin of this software must not be misrepresented; you must not
+ * claim that you wrote the original software. If you use this software in a
+ * product, an acknowledgment in the product documentation would be
+ * appreciated but is not required.
+ *
+ * 2. Altered source versions must be plainly marked as such, and must not be
+ * misrepresented as being the original software.
+ *
+ * 3. This notice may not be removed or altered from any source distribution.
+ *
+ * Ethan "flibitijibibo" Lee <flibitijibibo@flibitijibibo.com>
+ *
+ */
+
+#ifndef FNA3D_MEMORY_H
+#define FNA3D_MEMORY_H
+
+#include <SDL.h>
+
+/* Driver-defined memory structures  */
+
+typedef struct FNA3D_MemoryHandle FNA3D_MemoryHandle;
+typedef struct FNA3D_BufferHandle FNA3D_BufferHandle;
+
+/* Resource Access Types are mostly for setting up pipeline barriers */
+
+typedef enum FNA3D_Memory_ResourceAccessType
```

Comment:
This can probably go back in Vulkan.c; I thought it would be used more but really we just need to have a vertex/index buffer bool in CreateBufferContainer (and the associated driver function).

Reply:
I'm actually tempted to go the other direction - the renderer does need to indicate when barriers are needed, but I don't think it should necessarily be keeping track of the barrier state itself

Reply:
For now I've reverted this part, but if we need to expose it we can - this was only used to initialize vertex/index buffers; the barriers are handled within the Vulkan renderer without the allocator's involvement.